gui/login/static/signUp/SignUpController.js

```diff
@@ -28,6 +28,16 @@ angular
   $scope.progressMessage = '';
   $scope.showVerify = false;
   
+  $scope.enableOAuth = SigninUrl.indexOf("/oauth2/") !== -1;
+  $scope.signinLink = $scope.enableOAuth ? SigninUrl : '/login/' + ($rootScope.rd ? 'rd=' : '');
+  
+  // User should not be here if OAuth is enabled...
+  // Navigate them to the correct place
+  if ($scope.enableOAuth) {
+    $window.location.href = $scope.signinLink;
+    return;
+  };
+  
   // To handle special characters in passwords, we must escape them in the validation regex pattern
   // See https://stackoverflow.com/questions/3561493/is-there-a-regexp-escape-function-in-javascript/3561711#3561711
   $scope.escapeRegex = function(pattern) {
```

Comment:
Detect OAuth configuration in SignUpPage... Redirect if odd state encountered, disable page in case redirect doesn't work.